pyports/get_metadata.py

```diff
@@ -0,0 +1,165 @@
+        polygons_df['title'] = [d.get('title') for d in polygons_df.properties]  # add name column from nested dict
+        polygons_df['areaType'] = [d.get('areaType') for d in
+                                   polygons_df.properties]  # add areaType column from nested dict
+        polygons_df = polygons_df[['_id', 'geometry', 'title', 'areaType']]
+
+    polygons_df = polygons_df.rename(columns={'_id': 'polygon_id',
+                                              'areaType': 'polygon_area_type',
+                                              'title': 'name'})
+    polygons_df['geometry'] = polygons_df['geometry'].apply(shape)  # convert to shape object
+
+    polygons_df = gpd.GeoDataFrame(polygons_df)  # convert to GeoPandas
+
+    logging.info('polygons data extracted')
+
+    return polygons_df
+
+
+def get_vessels_info(import_path: str, db: pymongo.MongoClient = None, vessels_ids: List[str] = None) -> pd.DataFrame:
+
+    """
+    function that extract vessels info
+    :param import_path: path to the vessels info file location
+    :param db: MongoDB object
+    :param vessels_ids: filter data by specific vessels_ids
+    :return:
+    """
+
+    columns_projection = {"vesselId": 1, "class_calc": 1, "subclass_documented": 1, "built_year": 1,
+                          "name": 1, "deadweight": 1, "draught": 1, "size": 1, "age": 1}
+
+    # TODO: fill in mongo query
+    if not import_path:
+        col = db["vessels"]
+        col = col.find({'_id': {'$in': [ObjectId(vessel) for vessel in vessels_ids]}} if vessels_ids else {}, columns_projection)
+
+        vessels_df = pd.DataFrame(list(col)).drop(['_id'], axis=1)
+
+    else:
+
+        vessels_file_path = os.path.join(import_path, 'vessels.json')
+        with open(vessels_file_path, 'r') as vessels_file:
+            vessels_file = json.load(vessels_file)
+            vessels_df = pd.DataFrame.from_dict(vessels_file, orient='index')
+
+            if vessels_ids:
+                vessels_df = vessels_df[vessels_df.index.isin(vessels_ids)]
+
+            vessels_df = vessels_df[columns_projection].drop('vesselId', axis=1)
+
+    conditions = [(vessels_df["class_calc"] == "Cargo") & (vessels_df["subclass_documented"] == "Container Vessel"),
+                  (vessels_df["class_calc"] == "Cargo") & (vessels_df["subclass_documented"] != "Container Vessel"),
+                  (vessels_df["class_calc"] == "Tanker")]
+
+    choices = [VesselType.CARGO_CONTAINER.value, VesselType.CARGO_OTHER.value, VesselType.TANKER.value]
+    vessels_df["class_new"] = np.select(conditions, choices)
+    vessels_df["class_new"] = vessels_df["class_new"].replace({'0': 'other'})
+
+    vessels_df = vessels_df.add_prefix('vessel_')
+    vessels_df = vessels_df.reset_index().rename(columns={'index': 'vesselId'})
+
+    logging.info('vessels data extracted')
+
+    return vessels_df
+
+
+def get_ports_info(import_path: str, db: pymongo.MongoClient = None) -> pd.DataFrame:
```

Comment:
Seems like this function should not be a part of this file

Reply:
the methods in get_metadata are being used in more than one code.
in your opinion, where should we move it?